dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/plugins/OidcBaseClientValidator.java

```diff
@@ -92,10 +95,45 @@ public OidcBaseClient validateCreateUpdate() throws OidcServerException {
         return this.client;
     }
 
+    /**
+     * check for disallowed characters that could allow javascript to be fed back to ui.
+     */
+    private void detectIllegalChars() throws OidcServerException {
+        detectCh(client.getClientId());
+        detectCh(client.getClientSecret());
+        detectCh(client.getRedirectUris());
+        detectCh(client.getClientName());
+        detectCh(client.getPostLogoutRedirectUris());
+        detectCh(client.getPreAuthorizedScope());
+        detectCh(client.getFunctionalUserId());
+        detectCh(client.getFunctionalUserGroupIds());
+    }
+
+    // detect illegal chars
+    private void detectCh(@Sensitive String s) throws OidcServerException {
+        if (s == null || s.length() == 0) {
+            return;
+        }
+        for (int i = 0; i < illegalChars.length; i++) {
+            if (s.contains(illegalChars[i])) {
+                String description = TraceNLS.getFormattedMessage(OidcBaseClientValidator.class,
+                        MESSAGE_BUNDLE,
+                        "OAUTH_CLIENT_REGISTRATION_ILLEGAL_CHAR", // CWWKS1423E
+                        new Object[] { illegalChars[i] }, "");
+                throw new OidcServerException(description, OIDCConstants.ERROR_INVALID_CLIENT_METADATA, HttpServletResponse.SC_BAD_REQUEST);
+            }
+        }
+    }
+
+    // detect illegal chars
+    private void detectCh(@Sensitive JsonArray a) throws OidcServerException {
```

Comment:
The method should just be named `detectIllegalChars()`.

Reply:
seemed wordy but ok